packages/test/test-end-to-end-tests/src/test/messageSize.spec.ts

```diff
@@ -535,4 +535,96 @@ describeNoCompat("Message size", (getTestObjectProvider) => {
 			});
 		});
 	});
+
+	describe("Payload size on the wire", () => {
+		let totalPayloadSizeInBytes = 0;
+
+		const assertPayloadSize = (totalMessageSizeInBytes: number): void => {
+			// Expecting the message size on the wire should have
+			// at most 35% extra from stringification and envelope overhead.
+			const overheadRatio = 1.35;
+			assert.ok(
+				totalPayloadSizeInBytes < overheadRatio * totalMessageSizeInBytes,
+				`Message size on the wire, ${totalPayloadSizeInBytes} is larger than expected ${
+					overheadRatio * totalMessageSizeInBytes
+				}, after sending ${totalMessageSizeInBytes} bytes`,
+			);
+		};
+
+		const setup = async (containerConfig: ITestContainerConfig) => {
+			await setupContainers(containerConfig);
+			totalPayloadSizeInBytes = 0;
+			localContainer.deltaManager.outbound.on(
+				"push",
+				(messages) => (totalPayloadSizeInBytes += JSON.stringify(messages).length),
+			);
+		};
+
+		const compressionRatio = 0.1;
+		const badCompressionRatio = 1;
+		describe("Check batch size on the wire", () =>
+			[
+				{
+					messagesInBatch: 1,
+					messageSize: 1024,
+					expectedSize: 1 * 1024,
+					payloadGenerator: generateStringOfSize,
+				}, // One small uncompressed message
+				{
+					messagesInBatch: 3,
+					messageSize: 1024,
+					expectedSize: 3 * 1024,
+					payloadGenerator: generateStringOfSize,
+				}, // Three small uncompressed messages
+				{
+					messagesInBatch: 1,
+					messageSize: 1025 * 1024,
+					expectedSize: compressionRatio * 1025 * 1024,
+					payloadGenerator: generateStringOfSize,
+				}, // One large message with compression
+				{
+					messagesInBatch: 10,
+					messageSize: 1025 * 1024,
+					expectedSize: compressionRatio * 1025 * 1024,
+					payloadGenerator: generateStringOfSize,
+				}, // Ten large messages with compression
+				{
+					messagesInBatch: 10,
+					messageSize: 1025 * 1024,
+					expectedSize: badCompressionRatio * 10 * 1025 * 1024,
```

Comment:
I will add a comment here. What happens is that chunking works only when compression is enabled and kicks in if the compressed payload is large. In order to force that to happen, we need some really bad data (see the `payloadGenerator` property) so that compression kicks in, but it spits out a payload of roughly the same size. You usually get that when you compress random bits. This test configuration does just that. It creates 10 messages of roughly 1 MB each, and we expect the size on the wire to be roughly the same.